packages/schematics/angular/guard/index.ts

```diff
@@ -39,11 +39,18 @@ export default function (options: GuardOptions): Rule {
       .map(implement => implement === 'CanDeactivate' ? 'CanDeactivate<unknown>' : implement)
       .join(', ');
     let implementationImports = `${options.implements.join(', ')}, `;
+    const commonImports = 'ActivatedRouteSnapshot, RouterStateSnapshot, ';
+
+    implementationImports = `${implementationImports}${commonImports}`;
     // As long as we aren't in IE... ;)
     if (options.implements.includes(GuardInterface.CanLoad)) {
       implementationImports = `${implementationImports}Route, UrlSegment, `;
     }
 
+    if (options.implements.length === 1 && options.implements[0] === GuardInterface.CanLoad) {
```

Comment:
IMHO, I think this is a bit unnecessarily complicated. Something like the berlow, is easier to follow and cleaner.


```ts
const routerSnapshotsNameImports = ['ActivatedRouteSnapshot', 'RouterStateSnapshot'];
const routerNamedImports: string[] = [...options.implements];

if (options.implements.includes(GuardInterface.CanLoad)) {
  routerNamedImports.push(
    'Route',
    'UrlSegment',
    ...(options.implements.length > 0 ? routerSnapshotsNameImports : []),
  );
} else {
  routerNamedImports.push(...routerSnapshotsNameImports);
}

routerNamedImports.sort();
const implementationImports = routerNamedImports.join(', ');

```

Also, kindly add unit test.

Reply:
Thanks you for pointing  this out..It's much cleaner.

Reply:
Can you try to run

```
yarn bazel test //packages/schematics/angular:angular_test
```

Reply:
Are you getting an error when running `yarn bazel test //packages/schematics/angular:angular_test`?

Reply:
Yep..here the full output

![ng-cli-tests-output](https://user-images.githubusercontent.com/18737653/106648066-0fd14100-6590-11eb-8b71-8cff3aef7164.png)

Reply:
Can you try to rebase and check if the issue persist?

Reply:
I did a rebase and the errors does'nt appear anymore. 
by using `yarn bazel test //packages/schematics/angular:angular_test` i get now no errors.

i was expected to see some specs output in the console but the build only show success.

i tried to simulate some tests to failed, but i always get a success build... i'm missing something ? i went through the doc but found nothing to point me to the right way.

Thanks in advance

Reply:
> i was expected to see some specs output in the console but the build only show success

Only failed tests will appear in the terminal.

Note, the above command will only run the schematics angular tests. Which tests are amending?

Reply:
i change some tests inside Guard `index.spec.ts` just to see how a failed tests looks like. 
e.g

just change a line like : `expect(files).toContain('/projects/bar/src/app/foo/foo.guard.ts')` 

to : `expect(files).not.toContain('/projects/bar/src/app/foo/foo.guard.ts')`

and run the above command again. The build output always show success.

Reply:
I see, Bazel is running not in verbose failure mode. 

You can either check the errors in the log file which is pointed out in the above screenshot or run the test command with `--verbose_failures=true`.

Reply:
Ok thanks..i will take a look later on this..i'm AFK for the next 3 weeks.

Reply:
@alan-agius4 i had to switch to a linux subsystem to get tests running.

I just added some unit tests.